File: src/nl/rubensten/texifyidea/structure/LatexIncludePresentation.java
package nl.rubensten.texifyidea.structure;

import com.intellij.navigation.ItemPresentation;
import nl.rubensten.texifyidea.TexifyIcons;
import nl.rubensten.texifyidea.psi.LatexCommands;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.util.List;

/**
 * @author Ruben Schellekens
 */
public class LatexIncludePresentation implements ItemPresentation {

    private final String fileName;

    public LatexIncludePresentation(LatexCommands labelCommand) {
        if (!labelCommand.getCommandToken().getText().equals("\\include") &&
                !labelCommand.getCommandToken().getText().equals("\\includeonly") &&
                !labelCommand.getCommandToken().getText().equals("\\input")) {
            throw new IllegalArgumentException("command is no \\include(only)-command");
        }

        // Get label name.
        List<String> required = labelCommand.getRequiredParameters();
        if (required.isEmpty()) {
            throw new IllegalArgumentException("\\include(only) has no label name");
        }
        this.fileName = required.get(0);
    }

    @Nullable
    @Override
    public String getPresentableText() {
        return fileName;
    }

    @Nullable
    @Override
    public String getLocationString() {
        return "";
    }

    @Nullable
    @Override
    public Icon getIcon(boolean b) {
        return TexifyIcons.DOT_INCLUDE;
    }
}

File: src/nl/hannahsten/texifyidea/structure/latex/LatexIncludePresentation.kt
package nl.hannahsten.texifyidea.structure.latex

import com.intellij.navigation.ItemPresentation
import nl.hannahsten.texifyidea.TexifyIcons
import nl.hannahsten.texifyidea.psi.LatexCommands
import nl.hannahsten.texifyidea.reference.InputFileReference

/**
 * @author Hannah Schellekens
 */
class LatexIncludePresentation(labelCommand: LatexCommands) : ItemPresentation {

    private val fileName by lazy { labelCommand.references.filterIsInstance<InputFileReference>().joinToString { it.refText } }

    // This method has to be really quick because it is used for example in sorting entries before they are shown to the user in the structure popup, so we cannot resolve to the actual file here
    override fun getPresentableText() = fileName

    override fun getLocationString() = ""

    override fun getIcon(b: Boolean) = TexifyIcons.DOT_INCLUDE
}
